addons/sale_timesheet/models/product.py

```diff
@@ -19,6 +20,12 @@ class ProductTemplate(models.Model):
     project_id = fields.Many2one(domain="[('billable_type', '=', 'no'), ('allow_timesheets', 'in', [service_policy == 'delivered_timesheet' or '', True])]")
     project_template_id = fields.Many2one(domain="[('billable_type', '=', 'no'), ('allow_timesheets', 'in', [service_policy == 'delivered_timesheet' or '', True])]")
 
+    def unlink(self):
+        time_product = self.env.ref('sale_timesheet.time_product')
+        if time_product.product_tmpl_id in self:
+            raise ValidationError(_('This product cannot be archived, nor deleted.'))
```

Comment:
could be good to actually say which product is problematic (think if you remove 80 products)

Reply:
Good point ;)